File: tests/transpiler/x/kt/cross_join_filter.kt
fun main() {
    val nums = mutableListOf(1, 2, 3)
    val letters = mutableListOf("A", "B")
    val pairs = run {
    val _res = mutableListOf<MutableMap<String, Any>>()
    for (n in nums) {
        for (l in letters) {
            if ((n % 2) == 0) {
                _res.add(mutableMapOf("n" to n, "l" to l) as MutableMap<String, Any>)
            }
        }
    }
    _res
}
    println("--- Even pairs ---")
    for (p in pairs) {
        println(listOf(p["n"]!!, p["l"]!!).joinToString(" "))
    }
}

File: tests/transpiler/x/php/cross_join_filter.php
<?php
$nums = [1, 2, 3];
$letters = ["A", "B"];
$pairs = [];
foreach ($nums as $n) {
  foreach ($letters as $l) {
    if ($n % 2 == 0) {
      $pairs[] = ["n" => $n, "l" => $l];
    }
  }
}

echo rtrim("--- Even pairs ---"), PHP_EOL;
foreach ($pairs as $p) {
  echo rtrim($p["n"] . " " . $p["l"]), PHP_EOL;
}
?>
